tools/src/main/scala/org/apache/spark/sql/rapids/tool/AppFilterImpl.scala

```diff
@@ -88,41 +89,106 @@ class AppFilterImpl(
       apps
     }
 
-   // If `any` is specified as logicFilter config, then store the value in variable which is used
-   // later for final computation.
-   val appNameFilteredForDisjunction = if (filterAppName.nonEmpty && appArgs.any()) {
-     appNameFiltered
-   } else {
-     Seq.empty[AppFilterReturnParameters]
-   }
-
     val userNameFiltered = if (userName.nonEmpty) {
       val appNamelogicFiltered = if (appArgs.any()) {
         apps
       } else {
         appNameFiltered
       }
-      appNamelogicFiltered.filter(_.appInfo.exists(_.userName.contains(userName)))
+      appNamelogicFiltered.filter(_.appInfo.appStartInfo.exists(_.userName.contains(userName)))
     } else {
       appNameFiltered
     }
 
+    val configFiltered = if (filterSparkProperties.nonEmpty) {
+      val userNameLogicFiltered = if (appArgs.any()) {
+        apps
+      } else {
+        userNameFiltered
+      }
+
+      // check if key:value pair is provided in the sparkProperty filter. If those are provided
+      // along with just the key configs, then we have to make sure we consider both variations
+      // i.e key:pair and key configs to filter the event logs.
+      if (filterSparkProperties.toString.contains(":")) {
+        val individualConfigs = filterSparkProperties.map(str => str.split(":"))
+        val keyValueConfigs = individualConfigs.filter(kvConfig => kvConfig.size == 2)
+        val keysOnlyConfigs = individualConfigs.filter(kconfig => kconfig.size != 2).flatten(x => x)
+        val validConfigsMap = keyValueConfigs.map(a => a(0) -> a(1)).toMap
+
+        val configFilteredResult = userNameLogicFiltered.filter { appFilterReturnParameters =>
+          appFilterReturnParameters.appInfo.sparkProperties.exists { sparkProp =>
+            if (keyValueConfigs.nonEmpty || keysOnlyConfigs.nonEmpty) {
+              val allConfigs = sparkProp.configName // all configs from eventlog
+              //Intersection of configs provided in the filter args with event log configs.
+              val commonConfigs = validConfigsMap.keySet.intersect(allConfigs.keySet).map(
+                k => (k, allConfigs(k))).toMap
+
+              commonConfigs.keys.filter { key =>
+                commonConfigs(key) == validConfigsMap(key)
+              }.nonEmpty || filterSparkProperties.intersect(keysOnlyConfigs).nonEmpty
```

Comment:
You changed the wrong variables here.  If tests are passing we have a problem. 
isn't this intersecting the same thing, both of these comes from filterSparkProperties.  you should be intersecting with allConfigs.  My previous comment was to change filterSparkProperties to keysOnlyConfigs